src/saving/Save.cs

```diff
@@ -274,23 +251,9 @@ private static void WriteDataToSaveFile(string target, string justInfo, string s
 
         OutputEntry(tar, SAVE_INFO_JSON, Encoding.UTF8.GetBytes(justInfo));
 
-        if (tempScreenshot != null)
+        if (screenshot != null)
         {
-            byte[]? data = null;
-
-            using (var reader = new File())
-            {
-                reader.Open(tempScreenshot, File.ModeFlags.Read);
-
-                if (!reader.IsOpen())
-                {
-                    GD.PrintErr("Failed to open temp screenshot for writing to save");
-                }
-                else
-                {
-                    data = reader.GetBuffer((int)reader.GetLen());
-                }
-            }
+            byte[]? data = screenshot.SavePngToBuffer();
```

Comment:
This nullability seems wrong here. At least looking in Rider the Godot method signature is (notice how the return type is non-nullable):
```
public byte[] SavePngToBuffer() => NativeCalls.godot_icall_0_2(Image.method_bind_22, Object.GetPtr((Object) this));
```

Reply:
Yes, you are right. It should return empty array instead of null if anything goes wrong. Removed the "?" operator.